### ISSUE TITLE
File-Based Logging During Upgrade Ignores Kolibri Setting 

### ISSUE THREAD

**Issue by DXCanas** [learningequality/kolibri#9161]
## Observed behavior

Despite the `KOLIBRI_NO_FILE_BASED_LOGGING` envar being set, Kolibri tries to write to logs during upgrade.

## Errors and logs

```
Traceback (most recent call last):
  File "/usr/local/bin/kolibri", line 5, in <module>
    from kolibri.utils.cli import main
  File "/usr/local/lib/python3.6/site-packages/kolibri/utils/cli.py", line 17, in <module>
    from kolibri.plugins import config
  File "/usr/local/lib/python3.6/site-packages/kolibri/plugins/__init__.py", line 16, in <module>
    from kolibri.utils.conf import KOLIBRI_HOME
  File "/usr/local/lib/python3.6/site-packages/kolibri/utils/conf.py", line 47, in <module>
    os.mkdir(LOG_ROOT)
OSError: [Errno 30] Read-only file system: '/home/kolibri/.kolibri/logs'
```

## Expected behavior

Would assume that it Kolibri does not try to log to file when the `NO_FILE_BASED_LOGGING` setting is set to `True`

## Steps to reproduce

1. Set the `KOLIBRI_NO_FILE_BASED_LOGGING` envar to true
2. Run any command that triggers upgrade. `kolibri manage migrate` in my case

## Context

Was discovered during CD, in a Linux container of Ubuntu LTS 18.

**Comment by DXCanas** [comment]
Discovered by myself, mentioned to @rtibbles

**Comment by KshitijThareja** [comment]
Hi @rtibbles !
I looked into this issue a bit. The environment variable `KOLIBRI_NO_FILE_BASED_LOGGING` works as expected when explicitly typecasted to a boolean value in the code. 
The thing is, if the _'logs'_ folder doesn't already exist, the conf.py files tries to create it here:
https://github.com/learningequality/kolibri/blob/0f182a3ab253bc53c5ed8927e75f74f3e3b182f7/kolibri/utils/conf.py#L45-L47
Now, if the folder location as pointed by `KOLIBRI_HOME` is read-only, we'll always get the error mentioned in the issue. So, we just have to make sure that if the variable is set to True, we don't create the _'logs'_ folder, if it isn't already present.
Please correct me if I am wrong with my inference. 

Also, I'd like to work on this issue. Could you please assign it to me? I'd love to hear your suggestions on how to tackle it.

**Comment by rtibbles** [comment]
Excellent sleuthing @KshitijThareja! That does seem like the cause here. And I think you are absolutely correct, that we should only try to create the `LOG_ROOT` directory in the case that it does not exist and we do not have the `KOLIBRI_NO_FILE_BASED_LOGGING` flag set!

I have assigned you, looking forward to the pull request. I have retargeted the issue to the 0.16.0 milestone, as this would be a useful fix to have sooner rather than later and seems low risk. Please branch off release-v0.16.x and target there for your pull request, thank you!

**Comment by KshitijThareja** [comment]
Thanks @rtibbles! Will try to put up a PR soon for this 👍🏻

**Comment by rtibbles** [comment]
Fixed in #11758
